### PR TITLE
PLAT-935: Icon component

### PR DESCRIPTION
**Pull request by f-jauregui**
[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/PLAT-935)

#### Description
As per JIRA ticket

#### Important notes
- Updated icon usage across all stories that used at least one of them
- Updated VRT tests affected by the new component

#### ChangeLOG

- [ ] I have added all notable deployment/development environment (onprem!) changes to [CHANGELOG.md](https://github.com/reciprocity/zen-ui/blob/main/README.md)

#### Useful links
[Zen UI readme](https://github.com/reciprocity/zen-ui/blob/main/README.md)

[PR review guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/pull_request_reviews.md)

[Git commit guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/commit_guidelines.md)

### CODE REVIEW

**Review by Donaab — APPROVED** [review]

File: src/components/zen-icon/zen-icon.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  @Element() hostElement: HTMLZenIconElement;
+
+  /** Size of the icon. */
+  @Prop() readonly size: IconSizes = 'md';
+
+  /** Icon to be rendered. */
+  @Prop() readonly icon: IconDefinition = null;
+
+  render(): HTMLElement {
+    return <Host class={{ lg: this.size === 'lg' }}>{this.icon ? renderIcon(this.icon) : null}</Host>;
```
Comment (chavlji) [inline src/components/zen-icon/zen-icon.tsx:21]: Don't append size class. Use `reflect: true` and `:host([size='lg'])` in css.

File: src/components/zen-icon/zen-icon.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  styleUrl: 'zen-icon.scss',
+  shadow: true,
+})
+export class ZenIcon {
+  @Element() hostElement: HTMLZenIconElement;
+
+  /** Size of the icon. */
+  @Prop() readonly size: IconSizes = 'md';
+
+  /** Icon to be rendered. */
```
Comment (chavlji) [inline src/components/zen-icon/zen-icon.tsx:17]: Mention that we expect fa:
`Icon data (js file) imported from Font Awesome svg package`

File: src/components/zen-icon/zen-icon.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    element.icon = args.icon;
+  });
+  return html/*html*/ ` <zen-icon id="zen-icon-element" ...="${spreadArgs(args)}"></zen-icon> `;
+};
+
+# Icon
+
+Icon component wrapper
+
+## Component previews
```
Comment (chavlji) [inline src/components/zen-icon/zen-icon.stories.mdx:25]: Remove this if there're no special stories.

File: src/components/zen-icon/test/zen-icon.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+describe('zen-icon', () => {
+  it('should render with shadow dom', async () => {
+    const page = await newSpecPage({
+      components: [ZenIcon],
+      html: `<zen-icon></zen-icon>`,
+    });
+    expect(page.root.shadowRoot).toBeTruthy();
+  });
+
+  it('should apply the large class', async () => {
```
Comment (chavlji) [inline src/components/zen-icon/test/zen-icon.spec.tsx:14]: You won't need this test when switching from class to reflective attr.

File: src/components/zen-icon/test/zen-icon.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  });
+
+  it('should render a boxed icon', async () => {
+    const page = await newSpecPage({
+      components: [ZenIcon],
+      html: `<zen-icon></zen-icon>`,
+    });
+    page.rootInstance.icon = faUser;
+    page.waitForChanges();
+    expect(page.rootInstance.icon.prefix).toBe('fal');
```
Comment (chavlji) [inline src/components/zen-icon/test/zen-icon.spec.tsx:37]: Is this necessary? If you assign object to web-component, it should be assigned. It's not concern of this particular component.

File: src/components/zen-icon/types.tsx [hunk 6]
```diff
@@ -0,0 +1 @@
+export type IconSizes = 'md' | 'lg';
```
Comment (chavlji) [inline src/components/zen-icon/types.tsx:1]: Please remove this file and move all types to `components/helpers/types.ts`

File: src/components/zen-menu-item/zen-option.stories.mdx [hunk 7]
```diff
@@ -41,13 +44,20 @@ export const Slots = () => html/*html*/ `
     <div class="content">
       <img class="icon" src=${icon} alt="icon" />
       <b>Custom content</b>
-      <span style="margin-left: auto" .innerHTML="${litHtmlIcon(faCheck)}"></span>
+      <span style="margin-left: auto"><zen-icon id="icon1"></span>
```
Comment (chavlji) [inline src/components/zen-menu-item/zen-option.stories.mdx:47]: This looks way cleaner! Nice!

File: package.json [hunk 8]
```diff
@@ -103,6 +103,7 @@
     "sass-extract": "~2.1.0",
     "sass-extract-loader": "~1.1.0",
     "sass-loader": "~10.1.0",
+    "stencil-inline-svg": "^1.0.1",
```
Comment (chavlji) [inline package.json:106]: Where is this package used?

File: src/components/zen-button/zen-button.stories.mdx [hunk 9]
```diff
@@ -43,39 +43,40 @@ export const StoryLoading = () => html/*html*/ ` <zen-button loading>Button</zen
 
 export const StoryDisabled = () => html/*html*/ ` <zen-button disabled>Button</zen-button> `;
 
-export const StoryWithIcons = () =>
-  html/*html*/ `
+export const StoryWithIcons = args => {
+  ['#icon1', '#icon2', '#icon3', '#icon4', '#icon5'].forEach(id => {
```
Comment (chavlji) [inline src/components/zen-button/zen-button.stories.mdx:47]: Stories should be meaningful. Don't put same icon to all slots. Put cog or user icon to leading slot.

**Review by chavlji — APPROVED** [review]
A few minor improvements, but it looks great!

File: src/components/zen-icon/test/zen-icon.spec.tsx (hunk 5 above)
Comment (f-jauregui) [reply src/components/zen-icon/test/zen-icon.spec.tsx:37]: Shall I just remove it?

Comment (chavlji) [reply src/components/zen-icon/test/zen-icon.spec.tsx:37]: I would yeah. What do you think? Isn't it just testing that object is assigned to a web-component?

File: package.json (hunk 8 above)
Comment (f-jauregui) [reply package.json:106]: Removed. Was from a previous approach.
Thanks for this! 👍

File: src/components/zen-icon/test/zen-icon.spec.tsx (hunk 5 above)
Comment (f-jauregui) [reply src/components/zen-icon/test/zen-icon.spec.tsx]: Yeah, agree with you.

**Comment by f-jauregui** [comment]
@chavlji let's wait for #138 to be merged and then I'll update that story.

**Comment by chavlji** [comment]
@f-jauregui can we merge this?

**Pull request merged**